gdal/frmts/saga/sagadataset.cpp

```diff
@@ -406,16 +406,45 @@ GDALDataset *SAGADataset::Open( GDALOpenInfo * poOpenInfo )
 
 {
     /* -------------------------------------------------------------------- */
-    /*  We assume the user is pointing to the binary (i.e. .sdat) file.     */
+    /*  We assume the user is pointing to the binary (i.e. .sdat) file or a */
+    /*  compressed raster (.sg-grd-z) file.                                 */
     /* -------------------------------------------------------------------- */
-    if( !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sdat"))
+    if( !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sdat") &&
+        !EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sg-grd-z") )
     {
         return NULL;
     }
 
-    const CPLString osPath = CPLGetPath( poOpenInfo->pszFilename );
-    const CPLString osName = CPLGetBasename( poOpenInfo->pszFilename );
-    const CPLString osHDRFilename = CPLFormCIFilename( osPath, osName, ".sgrd" );
+    
+    CPLString osPath, osName, osHDRFilename; if (EQUAL(CPLGetExtension( poOpenInfo->pszFilename ), "sg-grd-z"))
+    {
+        const char* fileName = CPLGetFilename(poOpenInfo->pszFilename);
+        CPLString osPath("/vsizip/{");
+        osPath += fileName;
+        osPath += "}/";
+
+        char ** filesinzip = VSIReadDir(osPath);
+        if (filesinzip == NULL)
+            return NULL; //empty zip file
+        
+        CPLString file;
+        for (int iFile = 0; filesinzip != NULL && filesinzip[iFile] != NULL; iFile++)
+        {
+            if (EQUAL(CPLGetExtension(filesinzip[iFile]), "sdat"))
+            {
+                 file = filesinzip[iFile];
+                 break;
+            }
+        }
+        osName = osPath +  file;
+        osHDRFilename = CPLFormFilename (osPath, CPLGetBasename(file) , ".sgrd");
```

Comment:
just forget the above message. Just seeing that CPLFormFilename() works either with or without the starting dot character for the extension